src/main/java/com/fasterxml/jackson/databind/type/TypeFactory.java

```diff
@@ -260,11 +270,23 @@ public TypeFactory withCache(LRUMap<Object,JavaType> cache)  {
      * bigger maximum size.
      *
      * @since 2.12
+     * @deprecated Since 2.16. Use {@link #withCaches(CacheProvider)} instead.
      */
     public TypeFactory withCache(LookupCache<Object,JavaType> cache)  {
         return new TypeFactory(cache, _parser, _modifiers, _classLoader);
     }
 
+    /**
+     * Mutant factory method that will construct a new {@link TypeFactory}
+     * with cache instances provided by {@link CacheProvider}.
+     *
+     * @since 2.16
+     */
+    public TypeFactory withCaches(CacheProvider cacheProvider) {
```

Comment:
I think I'll change this so we won't need a new method -- caller can just construct `LookupCache` needed and pass that.